### PR TITLE
Add react spring

### PR DESCRIPTION
**Pull request by ehdlg**
## What type of PR is this? (Check all applicable)

- [x] 🚀 Feature
- [x] 🐛 Bug Fix
- [ ] 📚 Documentation
- [ ] 💻 Blog post
- [x] 🎨 Styles
- [ ] 🛠️ Refactoring
- [ ] 🕵️‍♀️ Code Review
- [ ] 🚑 Hotfix
- [x] 🧪 Test
- [ ] 🌐 Localization/Internationalization
- [ ] ⚙️  Configuration and Environment
- [ ] 🚫 Deprecation or Removal
- [ ] 📦 Release
- [ ] ⏪ Revert

## Description

**Please describe the changes in this pull request:**

Added animated counter numbers with react spring.
Fixed counter tests after the counter animations were added.

## Related Issues

- Addresses #160 

## Checklist

**Before submitting this pull request, please make sure you have done the following:**

### Code Quality

- [x] Reviewed your code for any potential issues.
- [x] Checked for coding style and formatting.

### Updated documentation?

- [ ] ✅ Yes
- [x] 🙅 No, because there is nothing to update
- [ ] 🙋‍♂️ No, I need help

### Added tests for your changes?

- [x] ✅ Yes
- [ ] 🙅 No, because they are not needed
- [ ] 🙋‍♂️ No, I need help

### UI Changes

- [ ] Added screenshots or recordings to demonstrate UI-related changes (if applicable).

## Additional Information

Replace this text with any additional information.

### CODE REVIEW

File: app/src/stores/pomodoro/constants.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type { StateType } from './types'
 
 export const DEFAULT_STATE_VALUES: StateType = {
   id: null,
-  counterContent: '25:00',
+  counterContent: '',
```
Comment (Sergih28) [inline app/src/stores/pomodoro/constants.ts:5]: What's the reason for this change? The store shouldn't change just because we are adding animations, no? :thinking:

**Review by Sergih28 — CHANGES_REQUESTED** [review]

File: app/src/stores/pomodoro/constants.ts (hunk 1 above)
Comment (ehdlg) [reply app/src/stores/pomodoro/constants.ts:5]: After reloading the page, there's a small delay where the animation changes to the default counter content, so I tried setting it to an empty string and I haven't seen any problem yet, but this is something we should check better.

**Comment by Sergih28** [comment]
We should use tailwind's `tabular-nums` class so that all digits have the same width, otherwise the counter moves slightly depending on the numbers it's showing.

File: app/src/stores/pomodoro/constants.ts (hunk 1 above)
Comment (Sergih28) [reply app/src/stores/pomodoro/constants.ts:5]: @ehdlg Can you look into this?
I think the animation shouldn't dictate to have an empty `counterContent` by default :thinking: 
The rest of the PR looks fine to me :+1:

**Comment by Sergih28** [comment]
Can you check again the things above please? @ehdlg 

I just build from scratch our docker containers and I still see the numbers "dancing" issue :sweat_smile:

File: app/src/stores/pomodoro/constants.ts (hunk 1 above)
Comment (Sergih28) [reply app/src/stores/pomodoro/constants.ts:5]: We agreed that this will be worked in another PR in the future.

**Review by Sergih28 — APPROVED** [review]
